sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -139,7 +139,7 @@ namespace Azure { namespace Storage { namespace Blobs {
       constexpr std::size_t BlockIdLength = 64;
       std::string blockId = std::to_string(id);
       blockId = std::string(BlockIdLength - blockId.length(), '0') + blockId;
-      return Internal::Base64EncodeText(blockId);
+      return Azure::Core::Base64Encode(std::vector<uint8_t>(blockId.begin(), blockId.end()));
```

Comment:
We should try to re-write this in terms of `vector<uint8_t>` and not have `blockId` be `std::string`. That way, you don't need to turn the string to vector of bytes when calling `Base64Encode`.

Reply:
Nah, `std::string` supports all kinds of operations like concatenation, to int, from int.
Ho can we convert `std::vectort<uint8_t>` between int in an easy way?